snmp/apache-stats.py

```diff
@@ -18,72 +18,68 @@
+        f.close()
 else:
-	# grab the status URL (fresh data)
-	# need debian package python-urlgrabber
-	from urlgrabber import urlread
-	data = urlread ( 'http://localhost/server-status?auto',
-					user_agent = 'SNMP Apache Stats' )
-	# write file
-	f = open ( CACHEFILE+'.TMP.'+`os.getpid()`, 'w' )
-	f.write ( data )
-	f.close()
-	os.rename ( CACHEFILE+'.TMP.'+`os.getpid()`, CACHEFILE )
+        # Grab the status URL (fresh data), needs package python-urlgrabber
+        from urlgrabber import urlread
+        data=urlread('http://localhost/server-status?auto', user_agent='SNMP Apache Stats').decode() # "data" is UTF string, need to decode.
+        # Write file
+        f=open(CACHEFILE+'.TMP.'+str(os.getpid()), 'w')
+        f.write(data)
+        f.close()
+        os.rename(CACHEFILE+'.TMP.'+str(os.getpid()), CACHEFILE)
 
 
 # dice up the data
-scoreboardkey = [ '_', 'S', 'R', 'W', 'K', 'D', 'C', 'L', 'G', 'I', '.' ]
-params = {}
+scoreboardkey=['_', 'S', 'R', 'W', 'K', 'D', 'C', 'L', 'G', 'I', '.']
+params={}
 for line in data.splitlines():
-	fields = line.split( ': ' )
-	if len(fields) <= 1:
-                continue  # "localhost" as first line cause out of index error
-	elif fields[0] == 'Scoreboard':
-		# count up the scoreboard into states
-		states = {}
-		for state in scoreboardkey:
-				states[state] = 0
-		for state in fields[1]:
-			states[state] += 1
-	elif fields[0] == 'Total kBytes':
-		# turn into base (byte) value
-		params[fields[0]] = int(fields[1])*1024
-	elif len(fields) > 1:
-		# just store everything else
-		params[fields[0]] = fields[1]
+        fields=line.split(': ')
+        if len(fields) <= 1:
+                continue  # "localhost" as first line causes out of index error
+        elif fields[0] == 'Scoreboard':
+            # count up the scoreboard into states
+            states={}
+            for state in scoreboardkey:
+                states[state]=0
+            for state in fields[1]:
+                states[state] += 1
+        elif fields[0] == 'Total kBytes':
+            # turn into base(byte) value
+            params[fields[0]]=int(fields[1])*1024
+        elif len(fields) > 1:
+            # just store everything else
+            params[fields[0]]=fields[1]
 
-# output the data in order (this is because some platforms don't have them all)
-dataorder = [
-	'Total Accesses',
-	'Total kBytes',
-	'CPULoad',
-	'Uptime',
-	'ReqPerSec',
-	'BytesPerSec',
-	'BytesPerReq',
-	'BusyWorkers',
-	'IdleWorkers'
+# output the data in order(this is because some platforms don't have them all)
+dataorder=[
+    'Total Accesses',
+    'Total kBytes',
+    'CPULoad',
+    'Uptime',
+    'ReqPerSec',
+    'BytesPerSec',
+    'BytesPerReq',
+    'BusyWorkers',
+    'IdleWorkers'
 ]
 for param in dataorder:
-	try:
-		print params[param]
-#                print param
-	except:	# not all Apache's have all stats
-		print 'U'
+    try:
+        print(params[param])
+    except: # not all Apache's have all stats
```

Comment:
i think  `except KeyError:`  is better to catch only not existing key - errors

Reply:
You mean change:
`except:  # not....`

to
`except KeyError:  # not...`
?